src/main/java/com/aerospike/documentapi/pathparts/MapPathPart.java

```diff
@@ -52,7 +52,10 @@ public Operation toAerospikeGetOperation(String binName, CTX[] contexts) {
     }
 
     @Override
-    public Operation toAerospikePutOperation(String binName, Object object, CTX[] contexts) {
+    public Operation toAerospikePutOperation(String binName, Object object, CTX[] contexts) throws IllegalArgumentException {
+        if (object.getClass().isArray()) {
+            throw new IllegalArgumentException("Unable to put an array, expecting a list or a map");
```

Comment:
It also can be a regular object.

Reply:
Yes, what I was trying to say is if a user tries to pass an array then he/she is likely to use a list or a map instead.

Reply:
Rephrased.